### PR TITLE
feat: add bom

### LINKED ISSUE #48: Add a BOM artifact
Please publish a google-cloud-bigtable-bom pom artifact that contains the synced versions of artifacts from this repository. We will consume them upstream in the google-cloud-bom as a single version rather than trying to keep 7+ artifact versions synced.

### PR DESCRIPTION
**Pull request by kolea2**
Fixes #48

### CODE REVIEW

File: google-cloud-bigtable-bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.google.cloud</groupId>
+    <artifactId>google-cloud-bigtable-bom</artifactId>
+    <version>1.6.1-SNAPSHOT</version><!-- {x-version-update:google-cloud-bigtable:current} -->
+    <packaging>pom</packaging>
+    <parent>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>google-cloud-shared-config</artifactId>
+        <version>0.2.1</version>
+    </parent>
```
Comment (igorbernstein2) [inline google-cloud-bigtable-bom/pom.xml]: I'm not sure if this is right.  This bom is meant to be imported by end users, but the shared config includes plugin configs  that are meant to be used by client developers

File: google-cloud-bigtable-bom/pom.xml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        </developer>
+        <developer>
+            <id>igorberstein</id>
+            <name>Igor Bernstein</name>
+            <email>igorbernstein@google.com</email>
+            <organization>Google LLC</organization>
+            <roles>
+                <role>Developer</role>
+            </roles>
+        </developer>
```
Comment (igorbernstein2) [inline google-cloud-bigtable-bom/pom.xml]: I think we should add Kristen here as well

File: google-cloud-bigtable-bom/pom.xml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <configuration>
+                    <skip>true</skip>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (igorbernstein2) [inline google-cloud-bigtable-bom/pom.xml]: newline please

File: google-cloud-bigtable-bom/pom.xml (hunk 1 above)
Comment (chingor13) [reply google-cloud-bigtable-bom/pom.xml]: Unless end users are using this artifact as their parent, they won't inherit the build rules in `google-cloud-shared-config`. We use that parent for all the release configuration (no compile/test/runtime dependency management in there)

Comment (igorbernstein2) [reply google-cloud-bigtable-bom/pom.xml]: Looks like other projects are using this pattern:
https://github.com/jersey/jersey/blob/faa809da43538ce31076b50f969b4bd64caa5ac9/bom/pom.xml#L48-L53
https://github.com/sonatype/oss-parents/blob/5cc24e62cae0fc8a1a6a917c581574b71dec476f/jvnet-parent/pom.xml

**Pull request merged**